### PR TITLE
Login flow

### PR DESCRIPTION
**Pull request by brucellino**
# Summary 

This describes a feature test for login flow to the forum using the EGI SSO IdP via CheckIn.
The feature is described at `tests/cucumber/features/login.feature`

----

**Related issue :** #28 #26 

**Consider author for inclusion**:
- [x] author list
- [x] contributor list

### CODE REVIEW

**Comment by brucellino** [comment]
The build failed because of the environment

File: tests/cucumber/features/login.feature [hunk 1]
```diff
@@ -1,5 +1,18 @@
 Feature: Log in with CheckIn A person with an identity can log in with CheckIn
```
Comment (NicolasLiampotis) [inline tests/cucumber/features/login.feature]: s/CheckIn/Check-in/g

File: tests/cucumber/features/login.feature [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+    And I am on the main page
+  
   Scenario: The login button is present
     Given I visit the main page
     When I look for the login button
     Then it is present
+
+  Scenario: A test user visits the forum to log in
+    Given I click on the login button
+    Then I am taken to EGI AAI
```
Comment (NicolasLiampotis) [inline tests/cucumber/features/login.feature]: I am taken to EGI AAI Check-in IdP Discovery page

File: tests/cucumber/features/login.feature [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+    Then I choose SSO
```
Comment (NicolasLiampotis) [inline tests/cucumber/features/login.feature]: In the current IdP Discovery page there is no dedicated "EGI SSO" login button so the user needs to use the search list. In the upcoming version there will also be a login button. Would it make sense to test both methods (i.e. testing if the login button-that-will-be-available-really-soon-now exists, as well as searching for it)?

Comment (brucellino) [reply tests/cucumber/features/login.feature]: We want to make sure that we understand user behaviour with these tests. Currently, the behaviour is to type something in the search box and see what the filtered list looks like. 
Implemented in the following way : 
```
@browser.text_field(id: 'username', name: 'j_username').set @username.to_s
```
If you add a new "EGI SSO" button, we should have a slightly different scenario where we write the feature and test it. 
I think the best thing is to first write the feature in [Gherkin](https://docs.cucumber.io/gherkin/) - how do you want users to behave?

**Pull request merged**